cmd/ci-secret-bootstrap/main.go

```diff
@@ -0,0 +1,247 @@
+			}
+			if secretContext.Namespace == "" {
+				return fmt.Errorf("found empty namespace value: '%v'", secretContext)
+			}
+			if secretContext.Name == "" {
+				return fmt.Errorf("found empty name value: '%v'", secretContext)
+			}
+			if _, ok := o.kubeConfigs[secretContext.Cluster]; !ok {
+				return fmt.Errorf("failed to find cluster context in the kubeconfig: '%s'", secretContext.Cluster)
+			}
+			secretContextString := fmt.Sprintf("%v", secretContext)
+			if toSet.Has(secretContextString) {
+				return fmt.Errorf("secret context '%s' showed more than once in the config", secretContextString)
+			}
+			toSet.Insert(secretContextString)
+		}
+	}
+	return nil
+}
+
+type Config = []SecretConfig
+
+type BWContext struct {
+	BWItem     string `json:"bw_item"`
+	Field      string `json:"field,omitempty"`
+	Attachment string `json:"attachment,omitempty"`
+}
+
+type SecretContext struct {
+	Cluster   string `json:"cluster"`
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
+
+type SecretConfig struct {
+	From map[string]BWContext `json:"from"`
+	To   []SecretContext      `json:"to"`
+}
+
+func loadConfig(path string) (Config, error) {
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	c := Config{}
+	err = yaml.Unmarshal(bytes, &c)
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func run(kubeConfigs map[string]rest.Config, config Config, bwClient bitwarden.BitwardenClient, dryRun bool,
+	secretClientGetter func(map[string]rest.Config, string) (coreclientset.SecretsGetter, error)) error {
+	if dryRun {
+		return fmt.Errorf("dryRun has not been implemented")
+	}
+	for _, secretConfig := range config {
+		data := make(map[string][]byte)
+		for key, bwContext := range secretConfig.From {
+			var value string
+			var err error
+			if bwContext.Field != "" {
+				value, err = bwClient.GetField(bwContext.BWItem, bwContext.Field)
+			} else {
+				value, err = bwClient.GetAttachment(bwContext.BWItem, bwContext.Attachment, true)
+			}
+			if err != nil {
+				return err
+			}
+			data[key] = []byte(value)
+		}
+		for _, secretContext := range secretConfig.To {
+			secret := &coreapi.Secret{
+				Data: data,
+				ObjectMeta: meta.ObjectMeta{
+					Name:   secretContext.Name,
+					Labels: map[string]string{"ci.openshift.org/auto-managed": "true"},
+				},
+			}
+			var secretsGetter coreclientset.SecretsGetter
+			secretsGetter, err := secretClientGetter(kubeConfigs, secretContext.Cluster)
+			if err != nil {
+				return err
+			}
+			_, err = secretsGetter.Secrets(secretContext.Namespace).Create(secret)
+			if err != nil {
+				if !kerrors.IsAlreadyExists(err) {
+					return err
+				}
+				if _, err := secretsGetter.Secrets(secretContext.Namespace).Update(secret); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func getSecretGetter(kubeConfigs map[string]rest.Config, cluster string) (coreclientset.SecretsGetter, error) {
```

Comment:
Why is this a function? Accessing a map seems pretty straightforward

Reply:
Done. With some workaround to make the `run` function testable.